static/webmention.js

```diff
@@ -96,255 +96,370 @@ A more detailed example:
-
-    // Shim i18next
-    window.i18next = window.i18next || {
-      t: function t(key) { return key; }
-    }
-    var t = window.i18next.t.bind(window.i18next);
-
-    function getCfg(key, dfl) {
-        return document.currentScript.getAttribute("data-" + key) || dfl;
+  "use strict";
+
+  // Shim i18next
+  window.i18next = window.i18next || {
+    t: function t(/** @type {string} */key) { return key; }
+  }
+  const t = window.i18next.t.bind(window.i18next);
+
+  /**
+   * Read the configuration value.
+   *
+   * @param {string} key The configuration key.
+   * @param {string} dfl A fallback value.
+   * @returns {string}
+   */
+  function getCfg(key, dfl) {
+    return document.currentScript.getAttribute("data-" + key) || dfl;
+  }
+
+  const refurl = getCfg("page-url", window.location.href.replace(/#.*$/, ""));
+  const addurls = getCfg("add-urls", undefined);
+  const containerID = getCfg("id", "webmentions");
+  /** @type {Number} */
+  const textMaxWords = getCfg("wordcount");
+  const maxWebmentions = getCfg("max-webmentions", 30);
+  const mentionSource = getCfg("prevent-spoofing") ? "wm-source" : "url";
+  const sortBy = getCfg("sort-by", "published");
+  const sortDir = getCfg("sort-dir", "up");
+  /** @type {boolean} */
+  const commentsAreReactions = getCfg("comments-are-reactions");
+
+  /**
+   * @typedef ReactEmoji
+   * @type {"in-reply-to"|"like-of"|"repost-of"|"bookmark-of"|"mention-of"|"rsvp"|"follow-of"}
+   */
+
+  /**
+   * Maps a reaction to a hover title.
+   *
+   * @type {Record<ReactEmoji, string>}
+   */
+  const reactTitle = {
+    "in-reply-to": t("replied"),
+    "like-of": t("liked"),
+    "repost-of": t("reposted"),
+    "bookmark-of": t("bookmarked"),
+    "mention-of": t("mentioned"),
+    "rsvp": t("RSVPed"),
+    "follow-of": t("followed")
+  };
+
+  /**
+   * Maps a reaction to an emoji.
+   *
+   * @type {Record<ReactEmoji, string>}
+   */
+  const reactEmoji = {
+    "in-reply-to": "💬",
+    "like-of": "❤️",
+    "repost-of": "🔄",
+    "bookmark-of": "⭐️",
+    "mention-of": "💬",
+    "rsvp": "📅",
+    "follow-of": "🐜"
+  };
+
+  /**
+   * @typedef RSVPEmoji
+   * @type {"yes"|"no"|"interested"|"maybe"|null}
+   */
+
+  /**
+   * Maps a RSVP to an emoji.
+   *
+   * @type {Record<RSVPEmoji, string>}
+   */
+  const rsvpEmoji = {
+    "yes": "✅",
+    "no": "❌",
+    "interested": "💡",
+    "maybe": "💭"
+  };
+
+  /**
+   * HTML escapes the string.
+   *
+   * @param {string} text The string to be escaped.
+   * @returns {string}
+   */
+  function entities(text) {
+    return text.replace(/&/g, "&amp;")
+      .replace(/</g, "&lt;")
+      .replace(/>/g, "&gt;")
+      .replace(/"/g, "&quot;");
+  }
+
+  /**
+   * Creates the markup for an reaction image.
+   *
+   * @param {Reaction} r 
+   * @param {boolean} isComment 
+   * @returns {string}
+   */
+  function reactImage(r, isComment) {
+    const who = entities(
+      (r.author && r.author.name) ? r.author.name : r.url.split("/")[2]
```

Comment:
This could probably be rewritten to use the [nullish coalescing operator](https://caniuse.com/mdn-javascript_operators_nullish_coalescing).

Reply:
I think it'd be:

```suggestion
      r.author?.name ?? r.url.split("/")[2]
```

Reply:
That looks about right (I tried different scenarios in the console).

Reply:
Note, that it requires an update of terser as per https://github.com/terser/terser/issues/830